### ISSUE TITLE
docker version

### ISSUE THREAD

**Issue by DAVIZINH0** [domoticz/domoticz#4616]
Hello,
I use docker version of domoticz since a couple of years. The linuxserver allways, and in the last year a old version (stable) because linuxserver dont update.

Today i see that exist a oficial docker image. My fault if this image was old.
But im very happy because a oficial image is great!!!
Im testing this with my old setup (a lot of scripts, integration xiaomi devices, mqtt, tasmotas, etc...)

Its posible to make sugestions to increase the usability of this oficial docker?
I have problem with the ssl port (configure in a variable).
other problem for me is that the time of system is configured as volume and synology not alow this. If i can set the time zone with a variable, this will be better.
Other interesting thing is launch a .sh with parameters in the first run of the docker after a update of the container.

i dont see info in the forum. sorry if this is not the correct place

**Comment by rwaaren** [comment]
You might want to open a new topic on the forum. If other users do face the same or similar issues and found solutions or work-arounds for it, they will post it as a reply. 
The audience on the forum is much wider then only the members seeing this github issue.

btw. I did see your earlier post on the forum but that is about a different question. (Which is the best version for docker?)

**Comment by DAVIZINH0** [comment]
yes, you are right. I will open a topic for this in the forum! thanks for your reply!

**Comment by gizmocuz** [comment]
@DAVIZINH0 , No problem to continue here.... as I get a email when there is a update!... 

The docker is quite new, So I am open for suggestions.

What is the issue with SSL ?

Does the docker not get the time from the system itself ? I would suspect this
Do you have an example of a docker that uses the timezone as a variable ?

What issue do you experience with the .sh with the parameters ?

**Comment by gizmocuz** [comment]
@DAVIZINH0 , If i have read correctly, I needed to add the tzdata package in the container.
I update the docker images, could you test with the new one ? (The Domoticz version could be the same, but hopefully the timezone is working now)

https://github.com/domoticz/domoticz-docker

**Comment by DAVIZINH0** [comment]
Hello gizmocuz, first: this is a great idea (the oficial docker, some people of spain use it and coment regulary in a telegram group). thanks for make it posible!!!

The ssl i think is my fault. yesterday (at night) i try some things but fail. today i will test again and i think it will works. Give me some time and i will tell you if there is a problem (yesterday the problem was I set the port 1443 in the variable and not works, only works 8443).

The time zone request, OK i read your recent post!. i will try today and tell you!

The .sh on startup i explain bad... very bad :-P
The feautre i need is launch a .sh at startup to install some librarys (sshpass, miio for a xiaomi humidifier, and other). In linuxserver docker exist a folder in de /config volume that you put a .sh and launch at startup. this was great for install this librarys.
In your domoticz docker, exist something similar to this?

thanks for this great idea, docker is very usefull for domoticz!!!

EDIT: timezone is working now
ssl working too!

**Comment by gizmocuz** [comment]
Great it is working. I also did some tests with the TZ environment, and it works indeed!
When you launch this docker, you should also have a config folder when using docker-compose, else I think you should have created a mapping to your Synology for the config folder correct?
You will for instance see the Domoticz database in there.
Which docker image did you used before ? I can try to perform the same action

**Comment by DAVIZINH0** [comment]
hello,
Previously i use the linuxserver image. https://hub.docker.com/r/linuxserver/domoticz/
They use a folder in /config called "custom-cont-init.d". If you put a .sh script in this folder. this script launch when the docker creates. This is usefull because if you upgrade the container, the sript execute and all works perfect without any modification
This container uses a diferent mapping folders. but it was easy migrate to your

I use synology assistant, and put the container with privileges and in host mode (usefull for xiaomi gateway), and then i map a some folders in my local storage, this is usefull because i use some scripts and other things with my 400 idx in domoticz :-)

This is my folder mapping, i tell you for your curiosity, no more :-)
1.- /dockers/domoticz_oficial/config ==>  /opt/domoticz/userdata/

- copy all the content to original styles to my styles folder, and then map it (usefull to load themes)
1.-/dockers/domoticz_oficial/styles ==> /opt/domoticz/www/styles/

- usefull for my ssl certificates and a script that i make to reload the certificate and dhl param in case of renoval in the router:
1.-/dockers/1_Comunes/Certificados ==> /opt/domoticz/certificados
2.-/dockers/domoticz_oficial/config/keys/server_cert.pem ==> /opt/domoticz/server_cert.pem

**Comment by DAVIZINH0** [comment]
2 more weeks of use and works perfect! the container is very stable!!
Only a problem with the oficial android app that report a error in domoticz log, but solved when I uninstall the android app.

I need a lot of librarys that i need to install manually (with a script) in the first launch.

I you whant i can show the scripts if its posible to add some of this librarys to the container

**Comment by gizmocuz** [comment]
@DAVIZINH0 , Great to hear. Yes please show me the script.
But is this Domoticz related ? What libraries do you need to install ?

**Comment by DAVIZINH0** [comment]
hello @gizmocuz 
I think this scripts maybe is unnecesary for the most people. i have a lot of things in my integration that maybe the people not need. maybe the more usefull is that the container launch a init.sh in a concret location always in the first start. and the users can put in this sh all the librarys they need. is a good idea? or a stupid idea? :-)

My script is this, maybe some things are duplicated because i use this in previous docker installation (linuxserver) and re-use
`#!/bin/bash

apt update

apt-get --assume-yes install gcc

apt-get --assume-yes install musl-dev

apt-get --assume-yes install python3-dev

apt-get --assume-yes install libffi-dev

apt-get --assume-yes install libssl-dev

apt-get --assume-yes install linux-headers

apt-get --assume-yes install grep

apt-get --assume-yes install sshpass

apt-get --assume-yes install mosquitto-clients

apt-get --assume-yes install python3-pip

apt-get --assume-yes install nano

apt-get --assume-yes install git

apt-get --assume-yes install wget

apt-get --assume-yes install curl

pip3 install -U setuptools

pip3 install -U virtualenv

python3 -m pip install -U pip

pip3 install -U python-miio

cd /opt/domoticz/plugins/xiaomi-mirobot

pip3 install msgpack-python

cd /opt/domoticz/scripts/personales/GHome

pip3 --no-cache-dir install https://github.com/bierdok/gnotify/archive/master.zip`

**Comment by gizmocuz** [comment]
Crazy bot we have here ;)

**Comment by DAVIZINH0** [comment]
@gizmocuz  do you change something important in the last buid?? 18/03/2021 ??
i deploy it and stop works some miio devices (xiaomi). i dont know if the problem is the docker side or in the miio side.
thanks!!
EDIT: the problem is miio side. i can check it!

**Comment by MuadDibVV** [comment]
Hello @gizmocuz,
You can find [here](https://blog.linuxserver.io/2019/09/14/customizing-our-containers/) the explanation of customization that should be interesting on your docker version.

**Comment by gizmocuz** [comment]
This has been implemented for quite some time now
